app/controllers/application_controller/filter.rb

```diff
@@ -947,7 +947,11 @@ def adv_search_build(model)
     # Restore @edit hash if it's saved in @settings
     @expkey = :expression                                               # Reset to use default expression key
     if session[:adv_search] && session[:adv_search][model.to_s]
-      @edit = copy_hash(session[:edit])
```

Comment:
@AparnaKarve I understand what caused the problem you are trying to solve here. But I still think that `@edit = copy_hash(session[:adv_search][model.to_s])` does not work for all the cases that might happen so it is not good to change it to what it was before. If it worked for all the cases, I wouldn't change it in my PR you have mentioned. Now I see, my solution was also not the best. So it's better to use your condition to prevent crashing some pages but it is also not the best solution. The best would be to find an answer to the question: when to use `session[:adv_search][model.to_s]` and when `session[:edit]` to avoid all the problems in all the pages. Unfortunately, we don't know it until it crashes because of `nil`. But I believe there is a pattern or some attribute which makes the difference. More time is needed for this problem.

From my point of view, I would just simplify your change and fix rubocop to be ready to be merged to make some pages work but after that, we should continue looking for the best solution.